source/kraken/apply_disruption.cpp

```diff
@@ -201,6 +195,149 @@ struct add_impacts_visitor : public apply_impacts_visitor {
         log_end_action(mvj->uri);
     }
 
+    void operator()(nt::disruption::LineSection& ls) {
+        std::string uri =
+            "line section (" +  ls.line->uri  + " : " + ls.start_point->uri + "/" + ls.end_point->uri + ")";
+        this->log_start_action(uri);
+        if (impact->severity->effect == nt::disruption::Effect::NO_SERVICE) {
+            LOG4CPLUS_TRACE(log, "canceling " << uri);
+
+            // Get all impacted VJs and compute the corresponding base_canceled vp
+            using SectionBounds = const std::pair<const nt::StopTime*, const nt::StopTime*>;
+            std::vector<std::tuple<const nt::VehicleJourney*, nt::ValidityPattern, SectionBounds>> vj_vp_pairs;
+
+            // Computing a validity_pattern of impact used to pre-filter concerned vjs later
+            // Loop on every (even partial) days of the period
+            type::ValidityPattern impact_vp{meta.production_date.begin()}; // bitset are all initialised to 0
```

Comment:
I think you can put this in a function in the impact no ?

it is already in `apply_disruption.cpp:212` and in this function we consider a day before the impact period in order to handle overmidnight impact. don't you think it could help you ?